File: data/gpt-4o/readable_format_HPEC/cpp/file252.cpp
#include <stdio.h>
#define DATA_PRINTF_MODIFIER "%f "

void print_array(int ni, int nl, float g[nl][ni]) {
 for (int i = 0; i < ni; i++) {
  for (int j = 0; j < nl; j++) {
   printf(DATA_PRINTF_MODIFIER, g[j][i]);
   if (((i * ni) + j) % 20 == 0)
    printf("
");
  }
 }
 printf("
");
}

File: data/llama3_8B_ins_results/readable_format_HPEC/fortran/file252.f90
subroutine print_array(ni, nl, g)
implicit none

DATA_TYPE, dimension(nl, ni) :: g
integer :: ni, nl
integer :: i, j
do i = 1, ni
do j = 1, nl
write(0, DATA_PRINTF_MODIFIER) g(j,i)
if (mod(((i - 1) * ni) + j - 1, 20) == 0) then
write(0, *)
end if
end do
end do
write(0, *)
end subroutine
